src/AutoRest.CSharp/Common/Input/CodeModelPartials.cs

```diff
@@ -275,189 +277,310 @@ internal partial class CodeModel
                 .Concat(Schemas.Groups);
     }
 
-    internal partial class TestDefinitionModel
+    internal partial class OavVariableScope
     {
-
-        [YamlMember(Alias = "useArmTemplate")]
-        public Boolean UseArmTemplate;
+        [YamlMember(Alias = "requiredVariables")]
+        public ICollection<string> RequiredVariables { get; set; } = Array.Empty<string>();
 
         [YamlMember(Alias = "requiredVariablesDefault")]
-        public System.Collections.Generic.Dictionary<string, string> RequiredVariablesDefault;
+        public Dictionary<string, string> RequiredVariablesDefault { get; set; }
+
+        [YamlMember(Alias = "secretVariables")]
+        public ICollection<string>? SecretVariables { get; set; }
+
+        [YamlMember(Alias = "variables")]
+        public Dictionary<string, object?> Variables { get; set; } = new Dictionary<string, object?>();
+
+        [YamlMember(Alias = "outputVariables")]
+        public Dictionary<string, string>? OutputVariables { get; set; }
+
+        public HashSet<string> GetAllVariableNames()
+        {
+            HashSet<string> variableNames = new HashSet<string>();
+            if (this.RequiredVariables is not null)
+            {
+                variableNames.UnionWith(this.RequiredVariables);
+            }
+            if (this.Variables is not null)
+            {
+                variableNames.UnionWith(this.Variables.Keys);
+            }
+            if (this.OutputVariables is not null)
```

Comment:
We could make this property non-nullable and if we do that, this check is not necessary